### PR TITLE
address multiple typo fixes.

### PR DESCRIPTION
**Pull request by fujitatomoya**
minor typo fixes.

Signed-off-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

### CODE REVIEW

File: docs/API/API.md [hunk 1]
```diff
@@ -472,7 +472,7 @@ the required operation.
 In non-blocking mode the only difference is that HANGUP, instead of blocking, makes
 the function exit immediately with an appropriate error code (such as SRT_EASYNC*,
 SRT_ETIMEOUT or SRT_ECONGEST) explaining why the function is not ready to perform
-the operation. Refer to the error descriptions in [API-funtions.md](API-funtions.md)
+the operation. Refer to the error descriptions in [API-functions.md](API-functions.md)
```
Comment (fujitatomoya) [inline docs/API/API.md:475]: this is actually broken link.

File: srtcore/buffer_rcv.cpp [hunk 2]
```diff
@@ -237,7 +237,7 @@ int CRcvBuffer::dropUpTo(int32_t seqno)
     // Move forward if there are "read/drop" entries.
     releaseNextFillerEntries();
     // Set nonread position to the starting position before updating,
-    // because start position was increased, and preceeding packets are invalid. 
+    // because start position was increased, and proceeding packets are invalid.
```
Comment (ethouris) [inline srtcore/buffer_rcv.cpp:240]: No, this is wrong. "Preceding" was meant here.

File: srtcore/buffer_rcv.cpp [hunk 3]
```diff
@@ -546,7 +546,7 @@ int CRcvBuffer::readBufferTo(int len, copy_to_dst_f funcCopyToDst, void* arg)
 
     // Update positions
     // Set nonread position to the starting position before updating,
-    // because start position was increased, and preceeding packets are invalid. 
+    // because start position was increased, and proceeding packets are invalid.
```
Comment (ethouris) [inline srtcore/buffer_rcv.cpp:549]: Idem line 240

File: srtcore/buffer_rcv.cpp [hunk 4]
```diff
@@ -984,7 +984,7 @@ int CRcvBuffer::scanNotInOrderMessageRight(const int startPos, int msgNo) const
 
 int CRcvBuffer::scanNotInOrderMessageLeft(const int startPos, int msgNo) const
 {
-    // Search preceeding packets to the left.
+    // Search proceeding packets to the left.
```
Comment (ethouris) [inline srtcore/buffer_rcv.cpp:987]: Idem line 240

File: srtcore/buffer_rcv.h [hunk 5]
```diff
@@ -142,7 +142,7 @@ class CRcvBuffer
     /// Query how many data has been continuously received (for reading) and available for reading out
     /// regardless of the TSBPD.
     /// TODO: Rename to countAvailablePackets().
-    /// @return size of valid (continous) data for reading.
+    /// @return size of valid (continuos) data for reading.
```
Comment (ethouris) [inline srtcore/buffer_rcv.h:145]: This isn't correct either, but I think the word meant here is "contiguous". @maxsharabayko ?

File: srtcore/buffer_rcv.h [hunk 6]
```diff
@@ -161,11 +161,11 @@ class CRcvBuffer
     /// Parameters (of the 1st packet queue, ready to play or not):
     /// @param [out] tsbpdtime localtime-based (uSec) packet time stamp including buffering delay of 1st packet or 0 if
     /// none
-    /// @param [out] passack   true if 1st ready packet is not yet acknowleged (allowed to be delivered to the app)
-    /// @param [out] skipseqno -1 or seq number of 1st unacknowledged pkt ready to play preceeded by missing packets.
+    /// @param [out] passack   true if 1st ready packet is not yet acknowledged (allowed to be delivered to the app)
+    /// @param [out] skipseqno -1 or seq number of 1st unacknowledged pkt ready to play proceeded by missing packets.
```
Comment (ethouris) [inline srtcore/buffer_rcv.h:165]: Idem, "preceded".

File: srtcore/buffer_rcv.h [hunk 7]
```diff
@@ -161,11 +161,11 @@ class CRcvBuffer
...
     /// @retval true 1st packet ready to play (tsbpdtime <= now). Not yet acknowledged if passack == true
     /// @retval false IF tsbpdtime = 0: rcv buffer empty; ELSE:
-    ///                   IF skipseqno != -1, packet ready to play preceeded by missing packets.;
+    ///                   IF skipseqno != -1, packet ready to play proceeded by missing packets.;
```
Comment (ethouris) [inline srtcore/buffer_rcv.h:168]: Idem line 165

File: srtcore/srt_attr_defs.h [hunk 8]
```diff
@@ -89,7 +89,7 @@ used by SRT library internally.
 // - Other compilers: none.
 ///////////////////////////////////////////////////////////////////////////////
 #if _MSC_VER >= 1920
-// In case of MSVC these attributes have to preceed the attributed objects (variable, function).
+// In case of MSVC these attributes have to proceed the attributed objects (variable, function).
```
Comment (ethouris) [inline srtcore/srt_attr_defs.h:92]: "precede" was meant here.

File: test/test_buffer_rcv.cpp [hunk 9]
```diff
@@ -658,7 +658,7 @@ TEST_F(CRcvBufferReadMsg, OnePacketTSBPD)
     EXPECT_FALSE(m_rcv_buffer->isRcvDataReady(m_tsbpd_base + m_delay));
 }
 
-/// TSBPD = ON, a ready to play packet is preceeded by a missing packet.
+/// TSBPD = ON, a ready to play packet is proceeded by a missing packet.
```
Comment (ethouris) [inline test/test_buffer_rcv.cpp:661]: preceded

File: srtcore/buffer_rcv.h (hunk 5 above)
Comment (maxsharabayko) [reply srtcore/buffer_rcv.h:145]: I meant "continuous" here, but missed one "i". :)

File: srtcore/buffer_rcv.h (hunk 6 above)
Comment (maxsharabayko) [reply srtcore/buffer_rcv.h:165]: This is more for @stevomatthews to judge. The intent was passive: "a packet preceded by missing packets".

File: srtcore/buffer_rcv.cpp (hunk 2 above)
Comment (stevomatthews) [reply srtcore/buffer_rcv.cpp:240]: Concur with @ethouris, but it is spelled "preceding" and refers to packets BEFORE the new start position.

File: srtcore/buffer_rcv.cpp (hunk 4 above)
Comment (stevomatthews) [reply srtcore/buffer_rcv.cpp:987]: preceding

File: srtcore/buffer_rcv.h (hunk 6 above)
Comment (stevomatthews) [reply srtcore/buffer_rcv.h:165]:
```suggestion
    /// @param [out] skipseqno -1 or sequence number of 1st unacknowledged packet (after one or more missing packets) that is ready to play.
```

File: srtcore/buffer_rcv.h (hunk 7 above)
Comment (stevomatthews) [reply srtcore/buffer_rcv.h:168]: preceded

File: test/test_buffer_rcv.cpp (hunk 9 above)
Comment (stevomatthews) [reply test/test_buffer_rcv.cpp:661]:
```suggestion
/// TSBPD = ON, a ready-to-play packet is preceded by a missing packet.
```

File: test/test_buffer_rcv.cpp [hunk 10]
```diff
@@ -696,7 +696,7 @@ TEST_F(CRcvBufferReadMsg, TSBPDGapBeforeValid)
     EXPECT_EQ(pkt_info.seqno, seqno);
     EXPECT_TRUE(pkt_info.seq_gap);
 
-    // The packet can't be read because there is a missing packet preceeding.
+    // The packet can't be read because there is a missing packet proceeding.
```
[the remark this answers is not in the document]
Comment (stevomatthews) [reply test/test_buffer_rcv.cpp:699]:
```suggestion
    // The packet can't be read because there is a missing packet preceding.
```

**Comment by stevomatthews** [comment]
Minor edits.

File: srtcore/buffer_rcv.cpp (hunk 2 above)
Comment (fujitatomoya) [reply srtcore/buffer_rcv.cpp:240]: I see, thanks for the explanation.

**Comment by fujitatomoya** [comment]
all comments addressed with https://github.com/Haivision/srt/pull/2570/commits/0333821a1262cc6294462d7779309edf687c7dc0.

CC: @stevomatthews Just FYI, i added you Co-Author on this since i just borrowed your suggestions.

**Review by maxsharabayko — APPROVED** [review]

**Pull request merged**